File: WasteVisionWebFE/src/data/mappers/objectPrediction-mapper.ts
import { ObjectPrediction } from "@/core/domain/ObjectPrediction";
import {
  ObjectPredictionDTO,
} from "../dto/objectPrediction-dto";

export class ObjectPredictionMapper {
  /**
   * Maps an ObjectPredictionDTO to a domain ObjectPrediction
   */
  public static toDomain(dto: ObjectPredictionDTO): ObjectPrediction {
    return {
      id: dto.id,
      x: dto.x,
      y: dto.y,
      width: dto.width,
      height: dto.height,
      category: dto.category,
      confidence: dto.confidence,
      predictionId: dto.predictionId,
    };
  }

  /**
   * Maps a domain ObjectPrediction to an ObjectPredictionDTO
   */
  public static toDTO(domain: ObjectPrediction): ObjectPredictionDTO {
    return {
      id: domain.id,
      x: domain.x,
      y: domain.y,
      width: domain.width,
      height: domain.height,
      category: domain.category,
      confidence: domain.confidence,
      predictionId: domain.predictionId,

    };
  }

  /**
   * Maps an array of ObjectPredictionDTOs to domain ObjectPredictions
   */
  public static toDomainList(dtos: ObjectPredictionDTO[]): ObjectPrediction[] {
    return dtos.map((dto) => this.toDomain(dto));
  }

  /**
   * Maps an array of domain ObjectPredictions to ObjectPredictionDTOs
   */
  public static toDTOList(domains: ObjectPrediction[]): ObjectPredictionDTO[] {
    return domains.map((domain) => this.toDTO(domain));
  }

}

File: WasteVisionWebBE/Domain/ObjectPrediction/ObjectPredictionMapper.cs
using System.Collections.Generic;
using System.Linq;
using DDDSample1.Domain.Predictions;
using DDDSample1.Domain.Shared;

namespace DDDSample1.Domain.ObjectPredictions
{
    public static class ObjectPredictionMapper
    {
        public static ObjectPredictionDto ToDto(ObjectPrediction obj)
        {
            if (obj == null) return null;

            return new ObjectPredictionDto(
                obj.Id.AsGuid(),
                obj.PredictionId.AsString(),
                obj.X.AsString(),
                obj.Y.AsString(),
                obj.Category.AsString(),
                obj.Confidence.AsString(),
                obj.Width.AsString(),
                obj.Height.AsString()
            );
        }

        public static List<ObjectPredictionDto> ToDtoList(List<ObjectPrediction> objs)
        {
            if (objs == null) return null;
            return objs.Select(o => ToDto(o)).ToList();
        }

        public static ObjectPrediction ToDomain(CreatingObjectPredictionDto dto)
        {
            if (dto == null) return null;

            return new ObjectPrediction(
                new PredictionId(dto.PredictionId),
                new NumberDouble(dto.X),
                new NumberDouble(dto.Y),
                new Description(dto.Category),
                new NumberDouble(dto.Confidence),
                new NumberDouble(dto.Width),
                new NumberDouble(dto.Height)
            );
        }
    }
}